compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -646,6 +646,11 @@ pub(crate) fn global_llvm_features(sess: &Session, diagnostics: bool) -> Vec<Str
         }
     }
 
+    // `+relaxed-simd` implies `+simd128`
+    if features.iter().any(|f| f == "+relaxed-simd") && !features.iter().any(|f| f == "+simd128") {
+        features.push("+simd128".into());
+    }
```

Comment:
Mind expanding this comment a bit? I think it'd be good to mention here:

* This code is wasm-specific
* Why this code is here and maybe a small snippet about how it relates to per-function features
* Maybe a quick mention of why this isn't generalized?

And then do you think it's worth adding a gate around this checking if the LLVM version is <= 19? That could link to the PR which changed this in upstream LLVM as well.

Reply:
Done! I hope its clear.
I also went ahead and renamed the whole system from "dependant" to "implicit", which is much more appropriate.

> And then do you think it's worth adding a gate around this checking if the LLVM version is <= 19? That could link to the PR which changed this in upstream LLVM as well.

I will keep an eye on it and make a PR as soon as it actually lands.